edgeworkers/examples/respond-from-edgeworkers/responseprovider/trace-headers/main.js

```diff
@@ -0,0 +1,114 @@
+/*
+(c) Copyright 2021 Akamai Technologies, Inc. Licensed under Apache 2 license.
+
+Version: 1.0
+Purpose:  Shows Forward request headers via responseProvider
+Repo: 
+Notes: If query param: getForwardHeaders=body exists, it will output all the headers as the response body instead
+      of the original body. Else it outputs them as response headers with a prefix: x-fwd-
+*/
+
+import { httpRequest } from 'http-request';
+import { createResponse } from 'create-response';
+import URLSearchParams from 'url-search-params';
+
+function constructResponseBody(request,response){
+  const lreq = Object.keys(request.getHeaders()).length
+  let t = "<html><body><h2>Request Headers:</h2><br>";
+  let i = 0;
+  while (i < lreq) {
```

Comment:
For iterating over arrays, could you instead make use of [Array forEach](
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)? Same goes for each other array iteration in this example. The result would look a lot cleaner and be easier to read.

In this instance you could instead do something like:

```
Object.keys(request.getHeaders()).forEach((key) => {
  request.getHeaders()[key].forEach((headerval) => {
    t += key + ": " + headerval + " <br>";
  });
});
```
(I haven't run this so apologies for minor syntax issues)